Configuration/AlCa/python/autoCond.py

```diff
@@ -31,16 +31,20 @@
     'run2_data_relval'             : '122X_dataRun2_relval_v1',
     # GlobalTag for Run2 HI data
     'run2_data_promptlike_hi'      : '122X_dataRun2_PromptLike_HI_v1',
-    # GlobalTag for Run3 HLT: it points to the online GT
-    'run3_hlt'                     : '122X_dataRun3_HLT_v4',
     # GlobalTag with fixed snapshot time for Run2 HLT RelVals: customizations to run with fixed L1 Menu
     'run2_hlt_relval'              : '122X_dataRun2_HLT_relval_v1',
+    # GlobalTag for Run3 HLT: it points to the online GT
+    'run3_hlt'                     : '122X_dataRun3_HLT_v4',
+    # GlobalTag with fixed snapshot time for Run3 HLT RelVals: customizations to run with fixed L1 Menu
+    'run3_hlt_relval'              : '122X_dataRun3_HLT_relval_v1',
     # GlobalTag for Run3 data relvals (express GT)
     'run3_data_express'            : '122X_dataRun3_Express_v3',
     # GlobalTag for Run3 data relvals
     'run3_data_prompt'             : '122X_dataRun3_Prompt_v3',
```

Comment:
```
----- Begin Fatal Exception 25-Apr-2022 19:22:53 CEST-----------------------
An exception of category 'ConditionDatabase' occurred while
   [0] Processing Event run: 346512 lumi: 250 event: 243042266
   [1] Running path 'dqmoffline_step'
Exception Message:
Payload of type LHCInfo with id 33134a07bf5346992723195b6b8ab9e21778b71d could not be loaded. An exception of category 'ConditionDatabase' occurred.
Exception Message:
De-serialization failed: the current boost version (1_75) is unable to read the payload. Data might have been serialized with an incompatible version. Payload serialization info:  {
"CMSSW_version": "CMSSW_12_3_0",
"architecture": "slc7_amd64_gcc10",
"technology": "boost/serialization",
"tech_version": "1_78"
 }
 from default_deserialize 
 from Session::fetchPayload 
----- End Fatal Exception -------------------------------------------------
```
in 138.4 which is by the way failing also in the [IBs]( https://cmssdt.cern.ch/SDT/cgi-bin/logreader/slc7_amd64_gcc900/CMSSW_12_2_X_2022-04-25-1100/pyRelValMatrixLogs/run/138.4_PromptCollisions+RunMinimumBias2021+ALCARECOPROMPTR3+HARVESTDPROMPTR3/step2_PromptCollisions+RunMinimumBias2021+ALCARECOPROMPTR3+HARVESTDPROMPTR3.log#/).
This only reinforces my opinion of *never* put infinite snasphot GlobalTag in the dictionary ;)

Reply:
I see your point, but I got convinced it worked when the HLT test I ran privately tourned out successful (see PR description https://github.com/cms-sw/cmssw/pull/37670#issue-1214772968).
Clearly I missed something, I'll do some more private testing and update the GT to a snapshot earlier in time if needed.

Reply:
> I got convinced it worked when the HLT test I ran privately tourned out successful 

yes, but after your changes the HLT tests are OK, but other consumers (e.g. relvals) are not. I'd just update the RelVal to use `run3_data_relval` as well. 
In the limit I'd rather remove all live workflow keys from autoCond which is essentially an offline beast, but maybe I am missing some use case (which should be handled with extreme care)

Reply:
> I'd just update the RelVal to use `run3_data_relval` as well.

In the past there was a "promptlike" GT, e.g. https://github.com/cms-sw/cmssw/blob/a5bbbf39d5767bc5a7aa462542fb886e735ed5ae/Configuration/AlCa/python/autoCond.py#L48
but that had infinite snapshot as well...
So maybe we could introduce a "prompt_relval" key, but I'm not sure it would make much sense...I think we need some discussion in @cms-sw/alca-l2. Of course any input is welcome!

> In the limit I'd rather remove all live workflow keys from autoCond which is essentially an offline beast, but maybe I am missing some use case (which should be handled with extreme care)

I see what you mean here, but it's also true that the HLT/Express/Prompt GTs were always included in the autoCond (as far as I can remember).

Reply:
> So maybe we could introduce a "prompt_relval" key, but I'm not sure it would make much sense...

the "prompt-like" keys were indeed as the Prompt GT, but with a finite snapshot, see e.g. the 10.6.X situation:

https://github.com/cms-sw/cmssw/blob/8906d3aa6e2ad266a2ad677154b7c992c2c4643e/Configuration/AlCa/python/autoCond.py#L37

(cf. https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/106X_dataRun2_PromptLike_v15)

> I see what you mean here, but it's also true that the HLT/Express/Prompt GTs were always included in the autoCond (as far as I can remember).

oh no, not at all, as you can see from the 10.6.X snapshot of the file, that's a recent invention...

Reply:
Fixed in 5306e7f to use fixed snapshot at 12_2_X last run, see (updated) PR description.